src/test/java/jp/ac/utokyo/rcast/karkinos/wavelet/WaveletDenoizeTest.java

```diff
@@ -0,0 +1,669 @@
+import jp.ac.utokyo.rcast.karkinos.exec.DataSet;
+
+public class WaveletDenoizeTest {
+    static Stream<Arguments> calcParameters() {
+        Map<String, Integer> dummyIndex = new HashMap<>();
+        @Jailbreak DataSet dataset1 = new DataSet(dummyIndex);
+        @Jailbreak DataSet dataset2 = new DataSet(dummyIndex);
+        dataset1.capintervalinit = true;
+        dataset2.capintervalinit = true;
+
+        double[] dummyCN1 = new double[3];
+        long[] NormalCnt11 = new long[] {1, 2, 3};
+        double[] TnRatio11 = new double[] {0.1, 0.2, 0.3};
+        long[] NormalCnt12 = new long[] {4, 5, 6};
+        double[] TnRatio12 = new double[] {1.1, 1.2, 1.3};
+
+        double[] dummyCN2 = new double[6];
+        long[] NormalCnt2 = new long[] {1, 2, 3, 4, 5, 6};
+        double[] TnRatio2 = new double[] {2.1, 2.2, 2.3, 2.1, 2.2, 2.3};
+
+        List<WaveletIF> list11 = createWiList(NormalCnt11, TnRatio11, dummyCN1, "chr1");
+        List<WaveletIF> list12 = createWiList(NormalCnt12, TnRatio12, dummyCN1, "chr2");
+        List<WaveletIF> list2 = createWiList(NormalCnt2, TnRatio2, dummyCN2, "chr1");
+        List<List<WaveletIF>> plist1 = new ArrayList<>(Arrays.asList(list11, list12));
+        List<List<WaveletIF>> plist2 = new ArrayList<>(Collections.singletonList(list2));
+
+        dataset1.capinterval = plist1;
+        dataset2.capinterval = plist2;
+
+        double[] expDeNoiseValue1 = new double[] {
+                0.21765567128029326, 0.21765567128029326, 0.21765567128029326,
+                1.206723294623865, 1.206723294623865, 1.206723294623865
+        };
+        double[] expCN1 = new double[] {0.5, 0.5, 0.5, 1.0, 1.0, 1.0};
+
+        double[] expDeNoiseValue2 = new double[] {
+                2.2109080498231073, 2.2109080498231073, 2.2109080498231073,
+                2.2109080498231073, 2.2109080498231073, 2.2109080498231073
+        };
+        double[] expCN2 = new double[] {7.0, 7.0, 7.0, 7.0, 7.0, 7.0};
+
+        return Stream.of(
+                //DataSet dataset, double expLoh, double[] expDeNoiseValue, double[] expCN
+                Arguments.of(dataset1, 0.51, expDeNoiseValue1, expCN1),
+                Arguments.of(dataset2, 0.90, expDeNoiseValue2, expCN2)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("calcParameters")
+    void calcTest(DataSet dataset, double expLoh, double[] expDeNoiseValue, double[] expCN) throws IOException {
+        double baselineLOHEstimate = WaveletDenoize.calc(dataset);
+        assertEquals(expLoh, baselineLOHEstimate, 0.000001);
+        List<List<WaveletIF>> capInterval = dataset.getCapInterval();
+        int cnt = 0;
+        for (List<WaveletIF> l : capInterval) {
+            for (WaveletIF wif : l) {
+                assertEquals(expDeNoiseValue[cnt], wif.getDenioseValue(), 0.000001);
+                assertThat(wif.getCN(), is(expCN[cnt]));
+                cnt = cnt + 1;
+            }
+        }
+    }
```

Comment:
I felt like a little bit hard to read the values tested so I want the data of interest to be completely separated from codes.
Something like the following will help to make it easy to add more test data.

```suggestion
    static Stream<Arguments> calcParameters() {
        return Stream.of(
            // normalCounts, tnRatios, expLoh, expDeNoiseValues, expCNs
            Arguments.of(
                new int[][] {{1, 2, 3}, {4, 5, 6}},
                new double[][] {{0.1, 0.2, 0.3}, {1.1, 1.2, 1.3}},
                0.51,
                new double[][] {{0.21765567128029326, 0.21765567128029326,
                    0.21765567128029326}, {1.206723294623865,
                    1.206723294623865, 1.206723294623865}},
                new double[][] {{0.5, 0.5, 0.5}, {1.0, 1.0, 1.0}}),
            Arguments.of(
                new int[][]{{1, 2, 3, 4, 5, 6}},
                new double[][] {{2.1, 2.2, 2.3, 2.1, 2.2, 2.3}},
                0.90,
                new double[][] {{2.2109080498231073, 2.2109080498231073,
                    2.2109080498231073, 2.2109080498231073,
                    2.2109080498231073, 2.2109080498231073}},
                new double[][] {{7.0, 7.0, 7.0, 7.0, 7.0, 7.0}}));
    }

    @ParameterizedTest
    @MethodSource("calcParameters")
    void calcTest(final int[][] normalCounts,
                  final double [][] tnRatios,
                  final double expLoh,
                  final double[][] expDeNoiseValues,
                  final double[][] expCNs) throws IOException {
        @Jailbreak final DataSet dataset = new DataSet(null);
        dataset.capintervalinit = true;
        dataset.capinterval = IntStream.range(0, tnRatios.length).mapToObj(i->
            IntStream.range(0, tnRatios[i].length).mapToObj(j->{
                final CapInterval ci = new CapInterval("chr"+i, j, j+1, true);
                ci.setCNVInfo(new CNVInfo(normalCounts[i][j],
                    0.0, 0.0, 1, 0.0, 0.0, tnRatios[i][j]));
                return (WaveletIF)ci;})
                .collect(Collectors.toList()))
            .collect(Collectors.toList());
        final double baselineLOHEstimate = WaveletDenoize.calc(dataset);

        final double delta = 1e-7;
        assertEquals(expLoh, baselineLOHEstimate, delta);
        final List<List<WaveletIF>> capInterval = dataset.getCapInterval();
        int chr = 0;
        for (final List<WaveletIF> l : capInterval) {
            assertArrayEquals(expDeNoiseValues[chr],
                l.stream().mapToDouble(WaveletIF::getDenioseValue).toArray());
            assertArrayEquals(expCNs[chr],
                l.stream().mapToDouble(WaveletIF::getCN).toArray());
            chr++;
        }
    }
```